cl/api/utils.py

```diff
@@ -91,6 +92,96 @@ def to_html(self, request, queryset, view):
+    `RelatedFilter` classes in filtersets. This is particularly useful for
+    scenarios where you need to filter on attributes of related models through
+    many-to-many relationships.
+
+    **Required Properties:**
+    - **`join_table_cleanup_mapping`**: A dictionary mapping specific lookups
+      or custom labels used for `RelatedFilter` fields to the corresponding
+      field names in the join table. This mapping is essential for correct
+      filtering.
+    """
+
+    join_table_cleanup_mapping: dict[str, str] = {}
+
+    def _get_filter_label(self: FilterSet, field_name: str) -> str:
+        """
+        Maps a filter field name to its corresponding label.
+
+        When defining filters(Declarative or using the `fields` attribute) in a
+        filterset, the field name used internally might not directly match the
+        the label used in the request. This method helps resolve this
+        discrepancy by mapping the given `field_name` to its correct label.
+
+        This is particularly useful for custom filter methods where only the
+        field name is available, and obtaining the triggering label is not
+        straightforward.
+
+        Args:
+            field_name (str): The field name as used within the filterset.
+
+        Returns:
+            str: The corresponding label for the given field name.
+        """
+
+        FIELD_NAME_LABEL_MAPPING = {
+            filter_class.field_name: label
+            for label, filter_class in self.filters.items()
+        }
+        return FIELD_NAME_LABEL_MAPPING[field_name]
+
+    def get_filters_for_join_table(
+        self: FilterSet, name: str
+    ) -> dict[str, Any]:
+        """
+        Processes request filters for use in a join table query.
+
+        Iterates through the request filters, cleaning and transforming them to
+        be suitable for applying filtering conditions to a join table. Returns
+        a dictionary containing the filtered criteria to be applied to the join
+        table query.
+
+        Args:
+            name: The name of label used to trigger the custom filtering method
+
+        Returns:
+            dict: A dictionary containing the filtered criteria for the join
+            table query.
+        """
+        filters = {}
+        filter_label = self._get_filter_label(name)
+        base_model_prefix = self._meta.model._meta.model_name
+        for filter_key, value in self.data.items():
+            # Skip custom filtering options triggered by the user
+            if filter_key.startswith(filter_label):
+                continue
+
+            cleaned_key = filter_key
+            # Add base model prefix for fields in Meta class
+            for basic_field, _ in self._meta.fields.items():
+                if cleaned_key.startswith(basic_field):
+                    cleaned_key = (
+                        f"{base_model_prefix}{LOOKUP_SEP}{filter_key}"
+                    )
```

Comment:
I noticed that this block of code is used to convert the base filters in the model (e.g., `id`, `date_created`, `date_modified`, `name` in the `attorney` model) into fields supported for nested lookups, such as` attorney__id`, `attorney__date_created`, `attorney__date_modified`, and `attorney__name`. I understand that this is necessary to prevent errors related to base fields not being supported for nested lookups.

However, if we apply the validation suggested in `filter_roles()` (in the other comment) to pass only valid `Role` fields to the query, could we remove these lines?

Since the base filters applied to nested instances might be redundant. For example, the attorney endpoint returns attorneys with nested `parties_represented`, which are `Roles` related to the same attorney. So any valid `attorney__` filter would already match all nested `Roles` if it matched the parent attorney?  And I think the same apply in the party endpoint.

Reply:
> the base filters applied to nested instances might be redundant

You're right. I'll add logic to ignore filter from the base model.